File: Examples/Mvc.VB/Controllers/FileManagerController.UsingPartial.vb
Imports GleamTech.AspNet.UI
Imports GleamTech.FileUltimate.AspNet.UI

Namespace Controllers
    Partial Public Class FileManagerController

        Public Function UsingPartial() As ActionResult
            Dim fileManager = GetFileManagerModel()

            Return View(fileManager)
        End Function

        Public Function FileManagerPartialView() As ActionResult
            Dim fileManager = GetFileManagerModel()

            Return PartialView(fileManager)
        End Function

        Private Function GetFileManagerModel() As FileManager
            Dim fileManager = New FileManager() With {
             .Width = CssLength.Percentage(100),
             .DisplayLanguage = "en"
            }
            fileManager.RootFolders.Add(New FileManagerRootFolder() With {
             .Name = "Root Folder 1",
             .Location = "~/App_Data/RootFolder1"
            })
            fileManager.RootFolders(0).AccessControls.Add(New FileManagerAccessControl() With {
             .Path = "\",
             .AllowedPermissions = FileManagerPermissions.Full
            })

            Return fileManager
        End Function

    End Class
End Namespace

File: Examples/AspNetCoreOnNetFullCS/Controllers/FileManagerController.UsingPartial.cs
using GleamTech.AspNet.UI;
using GleamTech.FileUltimate.AspNet.UI;
using Microsoft.AspNetCore.Mvc;

namespace GleamTech.FileUltimateExamples.AspNetCoreOnNetFullCS.Controllers
{
    public partial class FileManagerController
    {
        public IActionResult UsingPartial()
        {
            var fileManager = GetFileManagerModel();

            return View(fileManager);
        }

        public IActionResult FileManagerPartialView()
        {
            var fileManager = GetFileManagerModel();

            return PartialView(fileManager);
        }

        private FileManager GetFileManagerModel()
        {
            var fileManager = new FileManager
            {
                Width = CssLength.Percentage(100),
                DisplayLanguage = "en"
            };
            fileManager.RootFolders.Add(new FileManagerRootFolder
            {
                Name = "Root Folder 1",
                Location = "~/App_Data/RootFolder1"
            });
            fileManager.RootFolders[0].AccessControls.Add(new FileManagerAccessControl
            {
                Path = @"\",
                AllowedPermissions = FileManagerPermissions.Full
            });

            return fileManager;
        }
    }
}
